### PR TITLE
Deprecation Lines for BTPayPalNativeCheckout Module

### PR DESCRIPTION
**Pull request by KunJeongPark**
### Summary of changes

- Add deprecation message -
@available(*, deprecated, message: "BraintreePayPalNativeCheckout Module is deprecated, use BraintreePayPal Module instead") for all public classes and enums for BraintreePayPalNativeCheckout module

### Checklist

- [x] Added a changelog entry

- @KunJeongPark

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 ## unreleased
 * BraintreeCore
   * For analytics, only call `fetchOrReturnRemoteConfig()` when batch uploading, not on each analytic event enqueue
+* BraintreePayPalNativeCheckout
+  * Add deprecated warning in public classes and methods
```
Comment (jaxdesmarais) [inline CHANGELOG.md:7]:
```suggestion
* BraintreePayPalNativeCheckout (DEPRECATED)
  * **Note:** This module is deprecated and will be removed in a future version of the SDK
  * Add deprecated warning message to all public classes and methods
```

File: Demo/Application/Features/PayPalNativeCheckoutViewController.swift [hunk 2]
```diff
@@ -2,6 +2,7 @@ import Foundation
 import UIKit
 import BraintreePayPalNativeCheckout
 
+@available(*, deprecated, message: "BraintreePayPalNativeCheckout Module is deprecated, use BraintreePayPal Module instead")
```
Comment (jaxdesmarais) [inline Demo/Application/Features/PayPalNativeCheckoutViewController.swift:5]: This is our demo app, we don't need a deprecation message here, just in the public classes

Comment (KunJeongPark) [reply Demo/Application/Features/PayPalNativeCheckoutViewController.swift:5]: yes, but I got lint errors from referencing MXO classes here

File: CHANGELOG.md (hunk 1 above)
Comment (KunJeongPark) [reply CHANGELOG.md]: Addressed here: 59485d6ee2974840bd7803272d6b4c3bdb056e95

File: Demo/Application/Features/PayPalNativeCheckoutViewController.swift (hunk 2 above)
Comment (jaxdesmarais) [reply Demo/Application/Features/PayPalNativeCheckoutViewController.swift:5]: We don't have a linter on Braintree. Do you mean Cocoapods `pod lib lint`?

Comment (KunJeongPark) [reply Demo/Application/Features/PayPalNativeCheckoutViewController.swift:5]: yes. I think so.

Comment (KunJeongPark) [reply Demo/Application/Features/PayPalNativeCheckoutViewController.swift:5]: https://github.com/braintree/braintree_ios/actions/runs/9614991255/job/26521134948

Comment (KunJeongPark) [reply Demo/Application/Features/PayPalNativeCheckoutViewController.swift]: Looks ok without the line there! Are we ok with the lint warning here?

Comment (KunJeongPark) [reply Demo/Application/Features/PayPalNativeCheckoutViewController.swift]: addressed here: 991982b41a3dd722ba7c0c40ae1aab7861a0985a

**Review by scannillo — APPROVED** [review]

File: Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift [hunk 3]
```diff
@@ -37,6 +39,7 @@ import PayPalCheckout
         self.nativeCheckoutProvider = nativeCheckoutProvider
     }
 
+    @available(*, deprecated, message: "BraintreePayPalNativeCheckout Module is deprecated, use BraintreePayPal Module instead")
```
Comment (jaxdesmarais) [inline Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift:42]: We should move this one above the docstring instead of above the mark

Comment (KunJeongPark) [reply Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutClient.swift:42]: Great catch!

**Review by agedd — APPROVED** [review]
lgtm!

**Pull request merged**